### ISSUE TITLE
2.3 Issues Thread - master.lock: cannot open file: Permission denied

### ISSUE THREAD

**Issue by strarsis** [tomav/docker-mailserver#545]
The mail system starts up with `docker-compose up -d` - however, 
port 25 and 587 refuse a connection while other (e.g. 143 (IMAP)) work.
The postfix service isn't running at all, in container in 
'/var/log/mail/mail.err' (and '[...]mail.log') postfix logged an error:
````
postfix/master[1093]: fatal: open lock file /var/lib/postfix/master.lock: cannot open file: Permission denied
````
When this happens there should be at least an error logged to docker or startup cancelled.

**Comment by tomav** [comment]
Could you share your config and debug startup as stated [here](https://github.com/tomav/docker-mailserver/blob/master/CONTRIBUTING.md)? Thank you.

**Comment by strarsis** [comment]
The problem was caused by having different ownerships than root for the mounted host directories.

The issue is rather that the docker mail container fails silently instead failing at startup when postfix isn't able to start.

**Comment by tomav** [comment]
Want to add this check on startup?

**Comment by kiliant** [comment]
I just had this issue. Problem was that postfix couldn't start up somehow as ownership on `mailstate/lib-postfix/` was root:root and not 105:108.

Is there another way of restoring permissions than to remove the mailstate folder?

**Comment by tomav** [comment]
This may be added in `start-mailserver.sh` because we already fix some permissions if needed.
Other (additional) option is to have a standalone helper script that can be run on demand.
Note that upgrade to `ubuntu:16.04` may break user ids.

**Comment by zerpex** [comment]
Hi, I face the same issue since the last update, so I think it would be a really good idea to add a permission check at startup :)

regards

**Comment by Lordmau5** [comment]
Hey-ho!

I'll just pitch in with the same issue.

Had to downgrade to 2.3 to fix this, which is not optimal because there are some fixes in the latest build that I'd really like to have.

**Comment by tomav** [comment]
@kiliant any idea for adding this check?

**Comment by strarsis** [comment]
@tomav: One improvement would be already if Dovecot logs to container stdout/logs.
Using a process/service manager like supervisord or chaperone.d would allow proper exit in case of errors.

**Comment by tomav** [comment]
@strarsis: https://github.com/tomav/docker-mailserver/issues/434 We're open to PR

**Comment by kiliant** [comment]
@tomav an idea would be to integrate chown (or a permission check) in start-mailserver.sh. It is sufficient to chown from inside the container when the volume is mounted _rw_ (worked for me).

I would have to figure out what volume needs to be owned by what user and how to integrate that in the script. Can have a look at this if I have time, but maybe #586 has more prio.

**Comment by kortenkamp** [comment]
Just to put in a "me too" – I figured out that I have the same problem related to permissions in the mailstate volume. Can somebody give a list of correct ownerships for the lib folders here?

**Comment by tomav** [comment]
@kortenkamp permissions seem to work for me. Don't forget to start postgrey if not running.

**Comment by techgourmet** [comment]
This did solve the problem.
Adding a permissions check/fix on startup could prevent this as said by @kortenkamp

**Comment by tomav** [comment]
There is also `master.cf` to change, because `chroot` had `y` as default value. It's now `n`.

**Comment by tomav** [comment]
Here is the PR https://github.com/tomav/docker-mailserver/pull/612

**Comment by tomav** [comment]
`:latest` image available.

**Comment by thiesschneider** [comment]
I still cant send emails with this image. Do you have to use special envs?

thx!

**Comment by tomav** [comment]
What is the error in the logs when using `:latest`? It was a mess to reproduce so perhaps its not 100% fixed.

**Comment by thiesschneider** [comment]
I hat the :2.3 tag.
I will try again with the latest tag and post the logs if it does not work.

**Comment by tomav** [comment]
Thanks.

**Comment by tomav** [comment]
I'll update 2.3 tag once fix is confirmed.

**Comment by M-M-M-M** [comment]
Could you create a stable 2.4 tag with that fix? or a even 2.3.1?

**Comment by tomav** [comment]
@M-M-M-M thanks for confirming its working as expected.
I updated `2.3` tag (even if its not a good practice).
Currently building: https://hub.docker.com/r/tvial/docker-mailserver/builds/

**Comment by tomav** [comment]
Build available with `:2.3` tag.

**Comment by M-M-M-M** [comment]
Thank you but using the same tag is causing issue with caches as the new 2.3 is not refreshed until you removed it manually before any rebuild. For instance, I had to perform a `docker rmi 245b6fb89f3b` before getting a "correct" rebuild.

**Comment by kiliant** [comment]
Have you tried docker pulling with the 2.3 tag?

**Comment by M-M-M-M** [comment]
Of course. But I had already pulled the 2.3 tag version before the fix was applied. So once fixed has been deployed to same 2.3 tag then a new pull request of 2.3 was looking first in my local cache rather than on hub and so not retrieving the fixed 2.3 version.

**Comment by ChromoX** [comment]
I'm having this same problem on a brand new 2.3 tag.
